x-pack/legacy/plugins/infra/public/pages/metrics/components/sub_section.tsx

```diff
@@ -23,29 +23,25 @@ export const SubSection: FunctionComponent<SubSectionProps> = ({
   isLiveStreaming,
   stopLiveStreaming,
 }) => {
-  if (!children || !metrics) {
+  const metric = useMemo(() => metrics?.find(m => m.id === id), [id, metrics]);
+
+  if (!children || !metric) {
     return null;
   }
-  const metric = metrics.find(m => m.id === id);
-  if (!metric) {
+
+  const childrenWithProps = Children.map(children, child => {
+    if (isValidElement(child)) {
+      return cloneElement(child, {
+        metric,
+        id,
+        onChangeRangeTime,
+        isLiveStreaming,
+        stopLiveStreaming,
+      });
+    }
     return null;
-  }
-  const childrenWithProps = useMemo(
-    () =>
-      Children.map(children, child => {
-        if (isValidElement(child)) {
-          return cloneElement(child, {
-            metric,
-            id,
-            onChangeRangeTime,
-            isLiveStreaming,
-            stopLiveStreaming,
-          });
-        }
-        return null;
-      }),
-    [children, metric, id, onChangeRangeTime, isLiveStreaming, stopLiveStreaming]
```

Comment:
:information_source: Memoizing on `children` had no effect in this case because the parent always supplies a new list of React elements.